collections/src/main.rs

```diff
@@ -18,7 +23,7 @@ fn list_med(list: &[i32]) -> ListResult {
         for v in list {
             sum += *v;
         }
-        sum / len as i32
+        sum / len as i32 // TODO
```

Comment:
While code is OK, it feels unnecessarily imperative. Try to use `.fold()` here.

Reply:
I didn't know how to use predicates at that time. So I was just using imperative approaches to do one step of a time.